lib/hotel.rb

```diff
@@ -0,0 +1,108 @@
+require 'pry'
+
+module Hotel
+  require_relative 'room'
+  require_relative 'reservation'
+  require_relative 'room'
+  require_relative 'block'
+  require_relative 'exceptions'
+
+  class Hotel
+    attr_reader :rooms, :reservations, :blocks
+    ROOM_COST = 200
+
+    def initialize(num_rooms)
+      @rooms = []
+
+      unless num_rooms.class == Integer
+        raise(ArgumentError, "Number of rooms parameter must be Integer: #{num_rooms}")
+      end
```

Comment:
I usually recommend against this sort of argument checking. If the number of rooms isn't an integer, then an error will happen as soon as you get to the `times` call on the next line. Moreover, this prevents someone from implementing some other class that has a `times` method and using it here. [This is called "duck typing"](https://ericlippert.com/2014/01/02/what-is-duck-typing/), and it's one of the big ideas that make dynamic languages like Ruby nice to use.